vcd/resource_vcd_network_pool.go

```diff
@@ -0,0 +1,506 @@
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceNetworkPoolImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of network pool.",
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "Type of the network pool (one of `GENEVE`, `VLAN`, `PORTGROUP_BACKED`)",
+				ValidateFunc: validation.StringInSlice(supportedNetworkPoolTypes, false),
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the network pool",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Status of the network pool",
+			},
+			"promiscuous_mode": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Whether the network pool is in promiscuous mode",
+			},
+			"total_backings_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Total number of backings",
+			},
+			"used_backings_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Number of used backings",
+			},
+			"network_provider_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Id of the network provider (either vCenter or NSX-T manager)",
+			},
+			"network_provider_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the network provider",
+			},
+			"network_provider_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Type of network provider",
+			},
+			"backing": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Description: "The components used by the network pool",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"transport_zone": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Computed:    true,
+							MaxItems:    1,
+							Description: "Transport Zone Backing",
+							Elem:        resourceNetworkPoolBacking("resource"),
+						},
+						"port_groups": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Computed:    true,
+							MaxItems:    1,
+							Description: "Backing port groups",
+							Elem:        resourceNetworkPoolBacking("resource"),
+						},
+						"distributed_switches": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Computed:    true,
+							MaxItems:    1,
+							Description: "Backing distributed switches",
+							Elem:        resourceNetworkPoolBacking("resource"),
+						},
+						"range_ids": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "Distributed Switch ID ranges (used with VLAN)",
+							Elem:        resourceNetworkPoolVlanIdRange,
```

Comment:
We can't use `RequiredWith` because the check would fail when the backing element is missing (using `backing_components_use_constraint = "use-first-available"`).